File: src/test/java/com/wxl/dyttcrawler/processor/DyttGndyMoviePageProcessorTest.java
package com.wxl.dyttcrawler.processor;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.wxl.dyttcrawler.domain.DyttMovie;
import org.junit.Test;
import us.codecraft.webmagic.Page;
import us.codecraft.webmagic.Request;
import us.codecraft.webmagic.ResultItems;
import us.codecraft.webmagic.selector.PlainText;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.time.ZoneId;
import java.util.TimeZone;

/**
 * Create by wuxingle on 2020/5/10
 */
public class DyttGndyMoviePageProcessorTest {


    @Test
    public void test1() throws IOException {
        process("https://www.dytt8.net/html/gndy/dyzz/20200506/59996.html",
                "src/test/resources/detail1.html");
    }

    @Test
    public void test2() throws IOException {
        process("https://www.dytt8.net/html/gndy/dyzz/20200608/60105.html",
                "src/test/resources/detail2.html");

    }


    private void process(String urlstr, String html) throws IOException {
        Request request = new Request();
        request.setUrl(urlstr);
        request.setCharset("utf-8");
        request.setMethod("get");

        Page page = new Page();
        byte[] bytes = Files.readAllBytes(Paths.get(html));
        page.setBytes(bytes);
        page.setRequest(request);
        page.setUrl(new PlainText(urlstr));
        page.setDownloadSuccess(true);
        page.setCharset("utf-8");
        page.setRawText(new String(bytes, "utf-8"));
        page.setStatusCode(200);

        DyttGndyMoviePageProcessor processor = new DyttGndyMoviePageProcessor((p, url) -> 10, url -> true);

        processor.process(page);

        ResultItems resultItems = page.getResultItems();
        DyttMovie movie = resultItems.get(DyttMovie.class.getName());
        ObjectMapper mapper = new ObjectMapper();
        mapper.setTimeZone(TimeZone.getTimeZone(ZoneId.systemDefault()));
        System.out.println(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(movie));
        System.out.println(movie.getDesc());
    }
}

File: src/test/kotlin/com/wxl/dyttcrawler/processor/DyttGndyMoviePageProcessorTest.kt
package com.wxl.dyttcrawler.processor

import com.fasterxml.jackson.databind.ObjectMapper
import com.wxl.dyttcrawler.domain.DyttMovie
import com.wxl.dyttcrawler.downloadPage
import org.junit.jupiter.api.Test
import java.time.ZoneId
import java.util.*

/**
 * Create by wuxingle on 2021/10/12
 *
 */
class DyttGndyMoviePageProcessorTest {

    @Test
    fun test1() {
        process(
            "https://www.dytt8.net/html/gndy/dyzz/20200506/59996.html"
        )
    }

    @Test
    fun test2() {
        process(
            "https://www.dytt8.net/html/gndy/dyzz/20200608/60105.html"
        )
    }

    @Test
    fun test3() {
        process(
            "https://www.dytt8.net/html/gndy/jddy/20220130/62267.html"
        )
    }

    @Test
    fun test4() {
        process("https://dydytt.net/gndy/dyzz/20220103/62184.html")
    }

    private fun process(urlStr: String) {
        val page = downloadPage(urlStr)
        println(page.html)

        val processor = DyttGndyMoviePageProcessor(
            { _, _ -> 10 }, { true }
        )

        processor.process(page)

        val resultItems = page.resultItems
        val movie = resultItems.get<DyttMovie>(DyttMovie::class.java.name)

        val mapper = ObjectMapper().apply {
            setTimeZone(TimeZone.getTimeZone(ZoneId.systemDefault()))
        }

        if (movie != null) {
            println(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(movie))
            println(movie.desc)
        }
    }


}